tests/0330-oasis_C_Verify.phpt

```diff
@@ -0,0 +1,172 @@
+--TEST--
+OASIS Verify Basic test - C_VerifyInit(), C_Verify()
+--DESCRIPTION--
+Based on the same design than C_Sign() tests, let's first
+C_Sign() a data then let's C_Verify().
+--SKIPIF--
+<?php
+
+require_once 'require-module-load.skipif.inc';
+
+if (getenv('PHP11_MODULE') === false) {
+  echo 'skip';
+}
+
+$info = $module->getInfo();
+
+if (trim($info["manufacturerID"]) == 'SoftHSM')
+  echo 'skip: CKO_PRIVATE_KEY/CKK_RSA, missing with SoftHSM';
+
+if (getenv('PHP11_PIN') === false) {
+  echo 'skip';
+}
+
+?>
+--FILE--
+<?php declare(strict_types=1);
+
+$modulePath = getenv('PHP11_MODULE');
+$module = new Pkcs11\Module($modulePath);
+
+$rv = $module->C_GetSlotList(true, $s);
+var_dump($rv);
+
+$rv = $module->C_OpenSession($s[0], Pkcs11\CKF_SERIAL_SESSION, null, null, $session);
+var_dump($rv);
+var_dump($session);
+
+$pin = getenv('PHP11_PIN');
+if (strlen($pin) === 0)
+  $pin = null; # Smart card without any pin code
+
+$rv = $module->C_Login($session, Pkcs11\CKU_USER, $pin);
+var_dump($rv);
+
+$Template = [
+  Pkcs11\CKA_CLASS => Pkcs11\CKO_PRIVATE_KEY,
+  Pkcs11\CKA_KEY_TYPE => Pkcs11\CKK_RSA,
+//  Pkcs11\CKA_LABEL => 'xyz_PRIV_SIG', # the label of your private key
+];
+
+$rv = $module->C_FindObjectsInit($session, $Template);
+
+var_dump($rv);
+
+$rv = $module->C_FindObjects($session, $Objects);
+var_dump($rv);
+if (count($Objects) >= 1) {
+  echo "OK, got 1 Key" . PHP_EOL;
+} else {
+  die("Missing private RSA key");
```

Comment:
This assumes that there exists an RSA key prior to running this test. We should C_GenerateKeyPair() instead of trying to find an existing one.

Reply:
C_GenerateKeyPair() does not exist.